.circleci/config.yml

```diff
@@ -316,6 +321,8 @@ workflows:
       - smoke-php
       - smoke-python
       - smoke-ruby
+      # TODO(ethanbao): Separate the test and docker release into two jobs.
+      - build_and_push_docker_image
```

Comment:
Why is this being done twice? Also, without a branch filter like below, does this mean that docker images might get pushed from arbitrary branches?

Reply:
Good catch. The docker push should only happen for master branch.